src/js/components/app.coffee

```diff
@@ -187,20 +186,26 @@ App = React.createClass
       nowShowing: ShowingOverview
 
   setPage: (pageId) ->
-    @setState
-      pageId: pageId
-      nowShowing: ShowingOverview
-      selectedQuestion: null
-      selectedStudent: null
+    if pageId != @pageId()
+      @setState
+        nowShowing: ShowingOverview
+        selectedQuestion: null
+        selectedStudent: null
+
+  pageId: ->
+    parseInt(@props.params.pageId)
 
   getCurrentPage: ->
     pages = @state.sequence?.activities.map (a) -> a.pages
     pages = _.flatten pages
-    _.find pages, (p) => p.id == @state.pageId
+    _.find pages, (p) => p.id == @pageId()
 
   getQuestions: ->
     return [] unless @state.sequence
-    @getCurrentPage().questions || []
+    @getCurrentPage()?.questions || []
+
+  data: ->
+    Object.assign({}, @state, {pageId: @pageId()})
```

Comment:
Should you use _.assign here?  Object.assign isn't supported in IE 11 - http://kangax.github.io/compat-table/es6/#test-Object_static_methods_Object.assign